coreos-meta-translator/trans.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+
+from argparse import ArgumentParser, FileType
+from datetime import datetime
+import json
+import os
+
+
+def ensureDup(inp, out, inp_key, out_key):
+    if out.get(out_key, None) == None:
+        out[out_key] = inp.get(inp_key)
+        return out
+    if out.get(out_key) != inp.get(inp_key):
+        print("Input Files do not appear to be for the same release")
+        raise SystemExit(1)
+    return out
+
+def url_builder(stream, version, arch, path):
+    return f"https://fcos-builds.s3.amazonaws.com/prod/streams/{stream}/builds/{version}/{arch}/{path}"
+
+def get_extension(path, ext_count):
+    return ".".join(path.split(".")[-ext_count:])
+
+
+parser = ArgumentParser()
+parser.add_argument("--workdir", help="cosa workdir", required=True)
+parser.add_argument("--build-id", help="build id", required=True)
+args = parser.parse_args()
+
+outer_dir = os.path.join(args.workdir, "builds", args.build_id)
+release_file = os.path.join(outer_dir, "release.json")
+
+out = {}
+if os.path.exists(release_file):
+    with open(release_file, 'r') as w:
+        out = json.load(w)
+
+build_dirs = [os.path.join(outer_dir, arch_dir) for arch_dir in os.listdir(outer_dir) if os.path.isdir(os.path.join(outer_dir, arch_dir))]
+files = [os.path.join(build_dir, "meta.json") for build_dir in build_dirs if os.path.exists(os.path.join(build_dir, "meta.json"))]
+
+for f in files:
+    with open(f, 'r') as w:
+        input_ = json.load(w)
+
+        arch = input_.get("coreos-assembler.basearch")
+
+        out = ensureDup(input_, out, "buildid", "release")
+        out = ensureDup(input_.get('coreos-assembler.container-config-git'), out, 'branch', 'stream')
+
+        # build the architectures dict
+        arch_dict = {"media": {}}
+        arch_dict = ensureDup(input_, arch_dict, "ostree-commit", "commit")
+        generic_arches = [("aws", 2), ("qemu", 2), ("metal", 2), ("openstack", 2), ("vmware", 1)]
```

Comment:
Hmm, I wonder if we could just search for the first `.` instead? That way we don't have to encode these numbers here.

Reply:
With the current version scheme we're using `.` in the buildid (though that might be subject to change.

Reply:
Ahhh right! Hmm, OK how about just using the identifier for each artifact (e.g. `qemu`, `openstack`, ...) and get the extension after that? E.g.:

```
In [20]: x = 'fedora-coreos-30.340-qemu.qcow2.gz'

In [21]: x[x.index('qemu')+len('qemu'):]
Out[21]: '.qcow2.gz'
```

Reply:
Yeah, I can do that (I'll probably just go with an `rstrip` and anything that doesn't fit that mold can have it's own case [like the `installer` or `initramfs`])